umcontroller/umcontroller.go

```diff
@@ -18,36 +18,43 @@
 package umcontroller
 
 import (
+	"encoding/base64"
 	"fmt"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
 	"sort"
 	"sync"
 	"time"
 
 	"github.com/aoscloud/aos_common/aoserrors"
+	"github.com/aoscloud/aos_common/spaceallocator"
 	"github.com/aoscloud/aos_common/utils/cryptutils"
 	"github.com/looplab/fsm"
 	log "github.com/sirupsen/logrus"
 
 	"github.com/aoscloud/aos_common/api/cloudprotocol"
 	"github.com/aoscloud/aos_communicationmanager/config"
+	"github.com/aoscloud/aos_communicationmanager/fcrypt"
+	"github.com/aoscloud/aos_communicationmanager/fileserver"
 )
 
 /***********************************************************************************************************************
  * Types
  **********************************************************************************************************************/
 
-// URLTranslator translates local URL to remote.
-type URLTranslator interface {
-	TranslateURL(isLocal bool, inURL string) (outURL string, err error)
-}
-
 // Controller update managers controller.
 type Controller struct {
-	storage       storage
-	server        *umCtrlServer
-	urlTranslator URLTranslator
-	eventChannel  chan umCtrlInternalMsg
-	stopChannel   chan bool
+	storage      storage
+	server       *umCtrlServer
+	eventChannel chan umCtrlInternalMsg
+	stopChannel  chan bool
+	componentDir string
+
+	decryptContext Decrypter
```

Comment:
name it `decrypter`. It would be clear enough.

Reply:
Done